pkg/test/framework/components/istio/operator.go

```diff
@@ -513,6 +522,42 @@ func deployControlPlane(c *operatorComponent, cfg Config, cluster resource.Clust
 	return applyManifest(c, installSettings, istioCtl, cluster.Name())
 }
 
+func patchIngressPorts(cfg Config, cluster resource.Cluster) error {
```

Comment:
Why are we doing a patch instead of changing the main service?

Reply:
or is this temporary until we do the larger changes?

Reply:
See https://github.com/istio/istio/issues/25933#issuecomment-668152356; I'd like to just add the ports to the default list of ports but would rather wait till docs are ready. An alternative would be passing all ports, including the original defaults, to `ControlPlaneValues`

Reply:
To clarify, I'd intend to add to the default list of ports later

Reply:
I don't think we need to have all the docs, etc ready in order to make the change. `master` doesn't need to be perfect yet, we can temporarily break docs, etc. But ok to do it later

Reply:
@tbarrella could we do all of this through actual yaml under `samples/`? I'd like our tests to verify that the yaml actually works, rather than having its own way of exposing istiod.

Reply:
Sorry didn't see this earlier... I don't think this is worth it. I'm merging PRs in a certain order and this patch is only temporary:

1. Merge the Gateway sample into the istio repo (this PR)
2. Reference the sample in istio.io docs (after #1 so that the link isn't broken)
3. Add 15012 and 15017 to `ports` and remove this patch. Link to istio.io docs in the deprecation warning for meshExpansion